api/controllers/UserController.js

```diff
@@ -194,107 +137,111 @@ module.exports = {
-        promises.push(Ban.markTSVThreads(req.user.redToken, req.params.username));
+        promises.push(Ban.banFromSub(req.user.redToken, req.params.username, req.params.banMessage, req.params.banNote, 'pokemontrades', req.params.duration));
+        promises.push(Ban.banFromSub(req.user.redToken, req.params.username, req.params.banMessage, req.params.banNote, 'SVExchange', req.params.duration));
       }
-      promises.push(Ban.updateAutomod(req.user.redToken, req.params.username, 'pokemontrades', unique_fcs));
-      promises.push(Ban.updateAutomod(req.user.redToken, req.params.username, 'SVExchange', unique_fcs));
-      promises.push(Ban.addUsernote(req.user.redToken, req.user.name, 'pokemontrades', req.params.username, req.params.banNote));
-      promises.push(Ban.addUsernote(req.user.redToken, req.user.name, 'SVExchange', req.params.username, req.params.banNote));
-      promises.push(Ban.updateBanlist(req.user.redToken, req.params.username, req.params.banlistEntry, unique_fcs, igns, req.params.knownAlt));
-      promises.push(Ban.localBanUser(req.params.username));
+      if (!req.params.duration) {
+        if (flairs) {
+          promises.push(Ban.giveBannedUserFlair(req.user.redToken, req.params.username, flairs[0] && flairs[0].flair_css_class, flairs[0] && flairs[0].flair_text, 'pokemontrades'));
+          promises.push(Ban.giveBannedUserFlair(req.user.redToken, req.params.username, flairs[0] && flairs[1].flair_css_class, flairs[1] && flairs[1].flair_text, 'SVExchange'));
+          promises.push(Ban.markTSVThreads(req.user.redToken, req.params.username));
+        }
+        promises.push(Ban.updateAutomod(req.user.redToken, req.params.username, 'pokemontrades', unique_fcs));
+        promises.push(Ban.updateAutomod(req.user.redToken, req.params.username, 'SVExchange', unique_fcs));
+        promises.push(Ban.addUsernote(req.user.redToken, req.user.name, 'pokemontrades', req.params.username, req.params.banNote));
+        promises.push(Ban.addUsernote(req.user.redToken, req.user.name, 'SVExchange', req.params.username, req.params.banNote));
+        promises.push(Ban.updateBanlist(req.user.redToken, req.params.username, req.params.banlistEntry, unique_fcs, igns, req.params.knownAlt));
+        promises.push(Ban.localBanUser(req.params.username));
+      }
+      Promise.all(promises).then(function () {
+        console.log('Process to ban /u/' + req.params.username + ' was completed successfully.');
+        res.ok();
+      }, function(error) {
+        console.log(error);
+        res.status(error.statusCode || 500).json(error);
+      });
+      Event.create({
+        user: req.user.name,
+        type: "banUser",
+        content: "Banned /u/" + req.params.username
+      }).exec(function () {});
+    } catch (err) {
+      return res.serverError(err);
```

Comment:
I put this in a try-catch since it's async. Everything else is the same as before.